File: Frontend/src/app/models/profile.dto.ts
import { CompanyDto } from "./company.dto";
import { LanguageDto } from "./language.dto";
import { ProjectDto } from "./project.dto";
import { SkillDto } from "./skill.dto";

export interface ProfileDto {
  userId: string;
  skills: SkillDto[];
  languages: LanguageDto[];
  projects: ProjectDto[];
  companies: CompanyDto[];
}

File: Backend/Backend.Domain/DTOs/ProfileDto.cs
namespace Backend.Domain.DTOs;

public class ProfileDto
{
    public Guid UserId { get; set; }
    public List<SkillDto> Skills { get; set; }
    public List<LanguageDto> Languages { get; set; }
    public List<ProjectDto> Projects { get; set; }
    public List<CompanyDto> Companies { get; set; }
}
